### ISSUE TITLE
Beta Release

### ISSUE THREAD

**Issue by philwareham** [textpattern/textpattern-plugins-website#13]
OK, I'm fairly happy we can move to a beta shortly. The site is actually already up, as you've probably been aware:

https://plugins.textpattern.com/

I think the blockers to beta are:

- [x] Create accounts for you all in the admin-side (plus anyone else we feel would want to safely have access).
- [x] Create sitemap.xml TXP template.
- [x] Hammer out a few more kinks in the curated plugins list format and endpoints. [see here](https://github.com/textpattern/textpattern-curated-plugins-list/issues/7)
- [x] Add manifest flag to JSON output template.
- [x] Add dependencies flag(s) to JSON output template.
- [x] Add dependencies shortcode (TBC).
- [x] Ensure cards are created for the plugins that I've already added to the plugins site.
- [x] Ensure current pages in the site have correct information (e.g. verified versions, I just created a few random configurations just to test the filtering worked well enough).
- [x] Add more cards, plugin pages, and author pages just so the site isn't too bare at beta launch (more can obviously be added during beta test but I'd like to have around 50 plugins there for beta launch).
- [x] Create search console instances for Google, Bing, etc.
- [x] Set up site backups (@petecooper)

Feel free to test the current site and try to break it/spot typos/spot fundamental drawbacks of the system.

I've got to do some proper client work now so I'll be offline this afternoon mostly.

**Comment by Bloke** [comment]
Game on!

Just a side note, is there anything we can do about wrapping on underscore at intermediate widths?

![Wrapping plugin names](https://user-images.githubusercontent.com/21252/86465029-ac94dd00-bd28-11ea-9a12-acdfe7260e2c.png)

I had a quick interweb search and, back in 2011, it seems [the only CSS options](https://stackoverflow.com/questions/6316863/css-word-wrap-line-break-on-underscores-in-addition-to-whitespace-and-hyphens) are "break-all" which wraps on every character or to use JS to insert `<wbr>` around underscores. Have things moved on since then?

**Comment by philwareham** [comment]
> Just a side note: is there anything we can do about wrapping on underscore at intermediate widths?

Nope, it's just something we have to live with. `<wbr>` is still a thing but unless we can set it's behaviour somehow on that particular `<txp:title />` tag (say, whenever a `_` is used, it modifies it `_<wbr>` I don't see nice way off doing it. Certainly not via JS.

Not even sure that would look better though, this might be more confusing to end users:

```
wet_textfilter_
markdown
````

**Comment by Bloke** [comment]
Aye, true. No biggie.

**Comment by petecooper** [comment]
Hourly and daily backups provisionally done, will check in 24 hours to ensure cleanups are happening.

**Comment by philwareham** [comment]
OK, the site is now live and Textpattern navigation bar on all sites has been amended to point to it. I will create a forum topic to announce it.

Currently there are 33 plugins on the site just to start it off. Feel free to add additional cards to the curated plugins repo and I will then make pages for those in the site.

**Comment by Bloke** [comment]
Nice!

Are we going to leave .org around for the time being until we have more plugins on the new site? Is the plan to use redirect rules to migrate, e.g. https://textpattern.org/plugins/1186/smd_thumbnail to https://plugins.textpattern.com/plugins/smd_thumbnail., plus a blanket redirect to the home page for anything not represented?

**Comment by petecooper** [comment]
>Are we going to leave .org around for the time being until we have more plugins on the new site or start redirecting asap? 

Needs a full audit, that's a sizeable undertaking. I was considering proposing having it as a museum of sorts, though I haven't given it much thought yet. Combining .org content with the files in https://github.com/textpattern/textpattern-plugins-website/issues/9 to make a read-only archive might make for good crowdsourced project, if there's motivation from the community.

**Comment by Bloke** [comment]
Yeah we have Marios' database of plugins to trawl and audit in parallel. There may well be some gems in there that are still relevant but not represented on .org. Plus, I expect, a hole lot of dross that no longer runs or is no longer necessary.

**Comment by philwareham** [comment]
> Are we going to leave .org around for the time being

Yes, I have no immediate plans to take the .org site offline. I guess I could put a link through to it temporarily on the beta site so people can still find other plugins not currently in the new site.

Going forward, the museum idea is not bad - there are probably hundreds of older plugins on the .org site that are no longer needed and pointless to do the work bringing them all over. Note that the new site only supports plugins for Textpattern version 4.5 onwards anyway (I had two draw a line somewhere).

**Comment by petecooper** [comment]
>no longer runs or is no longer necessary.

Yeah, this. In two minds - hence my hesitation or lack of enthusiasm for proposing a museum: useful if there's a stupidly-old Textpattern site needing version _n_ of plugin _x_ to fix an immediate problem, I know I've certainly been there, but we need to be encouraging updates overall, I feel.

**Comment by petecooper** [comment]
>there are probably hundreds of older plugins on the .org site that are no longer needed and pointless to do the work bringing them all over

Yeah, heavy +1 to this. Nothing stopping someone else outside of core dev team taking this on, so might be worth gauging forum feedback if there's enough oomph.

**Comment by philwareham** [comment]
> my hesitation or lack of enthusiasm for proposing a museum

Another option would simply be to create a repo on GitHub to house files for all the really old plugins. No fancy explanations needed - just 'here are really old plugins stored for posterity, use at your own peril'.

**Comment by philwareham** [comment]
> Nothing stopping someone else outside of core dev team taking this on

I'm also hoping that people show enthusiasm for adding curated cards and then adding pages in the new plugin site for them (it's a really simple process, that I still need to document). I really don't want to be curating everything on my own (if at all). I'm just providing the tools, as it were.

**Comment by petecooper** [comment]
>'here are really old plugins stored for posterity, use at your own peril'.

Trivial to repurpose this - any good?: https://github.com/textpattern-community/textpattern-plugin-archive

**Comment by philwareham** [comment]
Forum announcement about plugins site being live: https://forum.textpattern.com/viewtopic.php?id=50877

**Comment by philwareham** [comment]
> Trivial to repurpose this - any good?

Yep, sounds like a plan! 😀
